ZenPacks/zenoss/ZenPackLib/tests/test_installed_zenpacks.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python
+
+##############################################################################
+#
+# Copyright (C) Zenoss, Inc. 2015, all rights reserved.
+#
+# This content is made available according to terms specified in
+# License.zenoss under the directory where your Zenoss product is installed.
+#
+##############################################################################
+"""Test this version of ZenPackLib against relevant installed ZenPacks"""
+
+import os
+from ZenPacks.zenoss.ZenPackLib.tests.ZPLTestHarness import ZPLTestHarness
+from ZenPacks.zenoss.ZenPackLib.tests.ZPLTestBase import ZPLTestBase
+
+
+class TestInstalledZenPacks(ZPLTestBase):
+    """Test this version of ZenPackLib against relevant installed ZenPacks"""
+
+    use_dmd = True
+    files = []
+
+    def afterSetUp(self):
+        super(TestInstalledZenPacks, self).afterSetUp()
+        for z in self.z.dmd.ZenPackManager.packs():
+            path = z.path()
```

Comment:
instead of opening and reading files you could just check to see if it's an instance of the ZenPackLib ZenPack:

```
from ZenPacks.zenoss.ZenPackLib.lib.base.ZenPack import ZenPack
if not isinstance(pack, ZenPack):
    continue
```

Check if it's an instance, then find and append the files, otherwise skip it